core/src/main/java/io/parsingdata/metal/data/selection/ByOffset.java

```diff
@@ -19,53 +19,60 @@
 import static io.parsingdata.metal.Util.checkNotNull;
 
 import io.parsingdata.metal.data.ParseGraph;
-import io.parsingdata.metal.data.ParseGraphList;
 import io.parsingdata.metal.data.ParseItem;
+import io.parsingdata.metal.data.ParseItemList;
 import io.parsingdata.metal.data.ParseValue;
+import io.parsingdata.metal.token.Token;
 
 public final class ByOffset {
 
     private ByOffset() {}
 
-    public static boolean hasGraphAtRef(final ParseGraph graph, final long ref) {
-        return findRef(ByType.getGraphs(graph), ref) != null;
+    public static boolean hasRootAtRef(final ParseGraph graph, final Token definition, final long ref) {
+        return findRef(ByToken.getAllRoots(graph, definition), ref) != null;
     }
 
-    public static ParseGraph findRef(final ParseGraphList graphs, final long ref) {
-        checkNotNull(graphs, "graphs");
-        if (graphs.isEmpty()) { return null; }
-        final ParseGraph res = findRef(graphs.tail, ref);
-        if (res != null) { return res; }
-        if (graphs.head.containsValue() && graphs.head.getLowestOffsetValue().getOffset() == ref) {
-            return graphs.head;
+    public static ParseItem findRef(final ParseItemList items, final long ref) {
+        checkNotNull(items, "graphs");
```

Comment:
I've decided to use this when working with a list such as `ParseItemList`, to clearly communicate that, as part of list processing, we're essentially _popping_ off the top value for handling.

In cases where we get a `ParseGraph` passed in, I prefer to use `graph.head` and `graph.tail` to communicate that we're traversing a data structure.